server/src/test/java/org/elasticsearch/action/admin/indices/mapping/get/GetMappingsResponseTests.java

```diff
@@ -84,22 +85,30 @@ protected GetMappingsResponse mutateInstance(GetMappingsResponse instance) throw
     public static ImmutableOpenMap<String, MappingMetaData> createMappingsForIndex() {
         // rarely have no types
         int typeCount = rarely() ? 0 : scaledRandomIntBetween(1, 3);
+        return createMappingsForIndex(typeCount, true);
+    }
+
+    public static ImmutableOpenMap<String, MappingMetaData> createMappingsForIndex(int typeCount, boolean randomTypeName) {
         List<MappingMetaData> typeMappings = new ArrayList<>(typeCount);
 
         for (int i = 0; i < typeCount; i++) {
-            Map<String, Object> mappings = new HashMap<>();
             if (rarely() == false) { // rarely have no fields
+                Map<String, Object> mappings = new HashMap<>();
                 mappings.put("field-" + i, randomFieldMapping());
                 if (randomBoolean()) {
                     mappings.put("field2-" + i, randomFieldMapping());
                 }
-            }
 
-            try {
-                MappingMetaData mmd = new MappingMetaData("type-" + randomAlphaOfLength(5), mappings);
-                typeMappings.add(mmd);
-            } catch (IOException e) {
-                fail("shouldn't have failed " + e);
+                try {
+                    String typeName = MapperService.SINGLE_MAPPING_NAME;
```

Comment:
I find it a little confusing that we can try to add the same type `_doc` multiple times, if `typeCount > 1`. Maybe this method should only take one parameter `boolean randomTypeName`, and then depending on its value we can choose a random `typeCount` within the method?

Reply:
> we can try to add the same type `_doc` multiple times

Those will be put into the same map entry in the `typeBuilder`though.

> Maybe this method should only take one parameter `boolean includeTypeName`, and then depending on its value we can choose a random `typeCount` within the method? If it's `false`, then we only allow 0 or 1 types, and if it's `true` we could have more.

Creating mapping responses for 0/1 type can be desired for and without random type name though at the moment. `GetMappingsResponseTests` currently tests both 0-3 types and random type names, while in `GetIndexResponse` I need only 0/1 type and cannot test with random names because we cannot send them over the rest layer anymore which is why I need "_doc" to make the equality comparison in the roundtrip tests work.

Reply:
Right, I was suggesting two cases based on a single parameter `randomTypeName`:
- If `randomTypeName == true`, then use a random type name and select 0 to 3 types.
- If `randomTypeName == false`, then only use `_doc` and select 0 or 1 types.

In any case, this is a very minor point, happy to do whatever you find cleanest.

Reply:
I'd rather clean that up when we remove all this temporary types/noTypes confusion alltogether.